aten/src/TH/generic/THStorage.cpp

```diff
@@ -142,28 +135,34 @@ THStorage* THStorage_(newWithDataAndAllocator)(real* data, ptrdiff_t size,
                                                THAllocator* allocator,
                                                void* allocatorContext) {
   THStorage *storage = static_cast<THStorage*>(THAlloc(sizeof(THStorage)));
```

Comment:
can you move this to non-generic?  (take a void * data and a backend).  It's better if all these pseudo-constructors are in the same file, because you end up changing them all together.

Reply:
I guess it doesn't matter because you can't move the free.

Reply:
I'll do this in a later patch, because if I move this I should also move the corresponding methods in THCStorage.cpp for consistency.